pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -106,6 +108,7 @@ type Config struct {
 		ApiKey     string `gcfg:"api-key"`
 		TenantId   string `gcfg:"tenant-id"`
 		TenantName string `gcfg:"tenant-name"`
+		TrustId    string `gcfg:"trust-id"`
```

Comment:
Regarding the code, indeed it is using a first token to get a new token scoped with the trust delegation, but this first token is automatically created from the full creds if specified.